### ISSUE TITLE
Question about delimiter detection's logic

### ISSUE THREAD

**Issue by mef** [mholt/PapaParse#156]
I'm wondering what is the purpose of the `avgFieldCount > 1.99` check in the delimiter detection code, more specifically where the value 1.99 comes from.

````javascript
if ((typeof bestDelta === 'undefined' || delta < bestDelta)
	&& avgFieldCount > 1.99)
````

 `avgFieldCount > 1` looks to make more sense to me, but I may be missing something.

The current code fails to detect delimiter on short CSV files with 2 columns.

**Comment by mholt** [comment]
I hacked that algorithm together on an airplane. It could probably use some analysis, but it generally does pretty well considering!

The `> 1.99` is just a more explicit way (IMO) of writing `>= 2`. A row with only 1 field (in other words, only 1 field) is of no use to delimiter detection. The algorithm has to tolerate rows which have different field counts, so the average field count may not always be an integer.

Also, does it fail on short CSV files with 2 columns? I've tried both:

```
a;b
```

and

```
a;b
c;d
```

which seem to work fine. (Granted, longer inputs are easier to extrapolate from...)
